### ISSUE TITLE
Umineko not starting on Ubuntu 18.04

### ISSUE THREAD

**Issue by ZahnPastor** [07th-mod/umineko-question#96]
i installed ubuntu 18.04 literally 2 days ago, set everything up etc, but the game refuses to launch with the patched executable

error message is 
`
System info: Intel CPU, with functions: MMX SSE SSE2 
Adding: .
Adding: ..
** Error ** Missing Game Data
No game script found. This application must be run from a directory containing NScripter, ONScripter, or Ponscripter game data.
`
original, unmodded umineko starts normally and is playable

furthermore i tried to attach the 0.utf with the `--script` parameter
the game "opens" then, but the screen is completely black, no sound plays nothing is interactable and the output nags me about not finding libpng12

now i have installed `libpng16`, `libpng-dev` and `libpng++-dev`, but it still doesnt work

i also tried reinstalled umineko and applying the patches again, leading to the same result

(i used the graphics and voice zips, as well as the update and the executable from the wiki, took the 0.utf from the install script, because that one is more up-to-date, but i've also tried the one provided in the wiki, with the same result)

**Comment by ItaloKnox** [comment]
I'm calling @Petsnew since he uses Linux. Unfortunately, I can't do much with only server knowledge.

**Comment by drojf** [comment]
Sorry, we updated the game so that it now launches the file 0.u instead of 0.utf. If you rename the script as `0.u` it should run.

The manual install instructions should tell you to rename the file, if it's not obvious I should update the documentation (I was also thinking about editing the .exe message so it tells you to rename the script to `0.u` ...I don't know why I didn't do that earlier)

**Comment by Petsnew** [comment]
Yeah, I was also having similar issues when I didn't rename `0.utf` to ´0.u´ and instead used the `--script` parameter. The game took to long to open and was quite buggy.

**Comment by ZahnPastor** [comment]
yeah ok, with having it being 0.u, it loads the script automatically, but the error with the script referencing and not finding libpng12 still persists. here's the log
`System info: Intel CPU, with functions: MMX SSE SSE2 
Adding: .
Adding: ..
Adding: /home/zepar/.Umineko4hdz/
Adding: ./:../
Couldn't initialize SDL gamecontroller: There are 0 joysticks available
Read global variables file
Error at line 12: unknown command [language]
(*define line 9)
Error at line 16: unknown command [errorsave]
(*define line 13)
Error at line 60: unknown command [addkinsoku]
(*define line 57)
Warning: alias 'on' may conflict with some barewords
Warning: alias 'off' may conflict with some barewords
Error at line 49666: unknown command [setlayer]
(*define line 49663)
Warning: alias 'white' may conflict with some barewords
Warning: alias 'black' may conflict with some barewords
Error at line 49676: unknown command [menu_click_def]
(*start line 1)
 *** can't load file [bmp\cursor0.png]: Failed loading libpng12.so.0: libpng12.so.0: cannot open shared object file: No such file or directory ***
 *** can't load file [bmp\cursor1.png]: Failed loading libpng12.so.0: libpng12.so.0: cannot open shared object file: No such file or directory ***
Error at line 314876: unknown command [menu_waveon]
(*check_game line 1)
 *** can't load file [bmp\background\forest\beach_2b.png]: Failed loading libpng12.so.0: libpng12.so.0: cannot open shared object file: No such file or directory ***
effect No. 50 is not implemented. Crossfade is substituted for that.
`

**Comment by drojf** [comment]
Can you try this old version of the .exe? It will use 0.utf instead of 0.u, so please put the filenames back:

https://github.com/07th-mod/umineko-question/files/1734237/Umineko5to8w_video.zip

**Comment by ZahnPastor** [comment]
i started it with that exe, the screen is still black, script complaining about libpng12, but i hear sound this time, i cant interact with any invisible menu items tho

**Comment by drojf** [comment]
I'll try to re-patch the fresh linux .exe  and get back to you, maybe something changed recently. Just for comparison, could you upload the .exe?

Also feel free to join our discord if you want to chat a bit easier

**Comment by ZahnPastor** [comment]
wait, i cant start the normal umineko game properly anymore too

**Comment by ZahnPastor** [comment]
ill try to redownload the original game again, and see if im a dummy and it may actually be mangagamers responsibility

**Comment by drojf** [comment]
I don't think so - most of the time the linux versions don't have these problems. And you managed to launch it fine the first time right? 

BTW, did you download from steam or from MangaGamer? (most of our players use steam I think)

**Comment by ZahnPastor** [comment]
Oh this is interesting, i can start the unmodded Umineko Version from steam,
i remember reading somewhere that steam brings it own runtime libraries so that games can just code against those, if i start the game in commandline, its not working properly

let me try to patch the game again, use the 0.u and start it through steam to see if that works now

**Comment by ZahnPastor** [comment]
ah, yes, ok i need to start the game through steam, because steam has libpng12 in their libraries.

so the whole debacle started because i didnt rename 0.utf to 0.u and then only proceeded to start the game through the commandline

make that line in the wiki extra big please, im the living example of overreading it and causing trouble for everyone involved

**Comment by drojf** [comment]
Don't worry, I was just about to re-upload the new .exes which remind you to rename the 0.utf as 0.u as the new error message instead of the generic/incorrect message.

I'll add that you have to start the game through steam (on linux), I didn't know that at all...
